src/librustdoc/doctest.rs

```diff
@@ -840,7 +840,7 @@ impl Collector {
         if !item_path.is_empty() {
             item_path.push(' ');
         }
-        format!("{} - {}(line {})", filename, item_path, line)
+        format!("{} - {}(line {})", filename.prefer_local(), item_path, line)
```

Comment:
@jyn514, would you be so kind and have another look if all the rustdoc changes seem OK to you?

Reply:
Below in `add_test()`, it tries to figure out a `path: PathBuf` using the return value of `get_filename()`. It becomes `doctest.rs` for anything other than a `Real` filename. If `get_filename()` were to return a PathBuf directly, should I make it `doctest.rs` too if it isn't a real file? That might change the behaviour here because it could be a `Custom("input")` (formatted as `<input>`)